src/ModelSingleton.py

```diff
@@ -191,26 +197,17 @@ def __call__(self, query_text, count):
             toks_cuda[k] = v.cuda(0)
         output = self.model(**toks_cuda)
         cls_rep = output[0][:, 0, :]
-        dist = cdist(cls_rep.cpu().detach().numpy(), self.all_reps_emb)
-        if count == 1:
-            nn_index = np.argmin(dist)
-            return [{
-                "label": self.all_reps_names[nn_index],
-                "curie": self.all_reps_ids[nn_index],
-                "distance_score": round(dist[0, nn_index], 3)
-            }]
-        count_dist = np.argpartition(dist, count, axis=None)
-        result_dist = np.sort(dist[0, count_dist[:count]], axis=None)
-        indices = [list(np.asarray(np.where(dist.flatten() == d)).flatten()) for d in result_dist]
-        indices = sum(indices, [])
-        return [
-            {
-                "label": self.all_reps_names[idx],
-                "curie": self.all_reps_ids[idx],
-                "distance_score": round(dist[0, idx], 3)
-            }
-            for idx in indices[:count]
-        ]
+        vector = cls_rep.cpu().detach().numpy().tolist()[0]
+        logger.info(f"Calculated Vector of {len(vector)} dims,")
+        logger.info("sending vector to elasticsearch")
+        return await self.elastic_client.search(
+                query_vector=vector,
+                top_n=count,
+                bl_type=bl_type,
+                algorithm=similarity
+            )
+
```

Comment:
Nice simplified code to use elastic search for searching, but as I mentioned in my previous comment, it is probably beneficial to have a configuration variable to indicate whether to use elastic search backend or not so we can keep both elastic search and in-memory search setup for supporting different use cases.